Post Auth Merge Fixes

shaikh-rashid opened the pull request:
WIP Draft PR

Commit:
fixing authx-down command

Commit:
container name patches

Commit:
container_name cleanup

Commit:
DIG-515 : authentication refactoring

Commit:
DIG-515: authz

Commit:
DIG-512

Commit:
DIG-513

Commit:
DIG-510

Commit:
begin authx setup skipping local idp

Commit:
DIG-511 + external compose volumes

shaikh-rashid committed:
Merge branch 'develop' into hotfix/post-auth-merge-fixes and refactoring of aai

shaikh-rashid committed:
Merge branch 'develop' into hotfix/post-auth-merge-fixes and refactoring of aai

amanjeev committed:
update (conda): settings that allow for conda env setup without intervention

DIG-633

amanjeev committed:
add (vagrant): libvirt section to launch using stuff like QEMU

DIG-633

amanjeev commented on Makefile:
```diff
@@ -350,148 +380,6 @@ compose:
 	# 	| docker-compose -f - up
 
 
-#TODO: deprecate compose-authx-down
```
is this deprecated but still saved in `Makefile.auth.mak` ?

amanjeev commented on docs/install-docker.md:
````diff
@@ -96,17 +96,16 @@ sudo usermod -aG docker $(whoami)
 ```bash
 # 1. initialize repo and submodules
-git clone -b stable https://github.com/CanDIG/CanDIGv2.git
+git clone -b develop https://github.com/CanDIG/CanDIGv2.git
 git submodule update --init --recursive
+cd CanDIGv2
````
Do we want `develop` branch for installation?

Also, I think 101 should be before 100.  I believe we have to `cd` into the cloned repo and then `submodule init`.

amanjeev commented:
@shaikh-rashid Is [docs/authz-setup.md](docs/authz-setup.md) still the way to deploy this setup?  It talks about `compose-authx-setup-candig-server` which is deprecated, right?

The other thing is that this seems to be deploying `candig-server` which is v1, right?  Should get rid of that in this repo and use the other services that should live behind tyk?

I am just trying to figure out how to deploy auth stuff now.  I see there are two separate directories - authorization and authentication - which is perhaps a change from using `authz` dir.

amanjeev committed:
fix (conda): removes hard-coded instances of CONDA, uses single CONDA to avoid edge case  DIG-633

amanjeev committed:
refactor (conda): use common variable for CONDA path

amanjeev committed:
Add Authentication Tools - Tyk and Keycloak (#99)

* update (conda): settings that allow for conda env setup without intervention

DIG-633

* add (vagrant): libvirt section to launch using stuff like QEMU

DIG-633

* refactor (authx): reorganizing structure; WIP; DIG-633

* chore (README): spelling

* chore (gitignore): add .idea directory

* feature (vagrant): add IP address option to Vagrantfile

* feature (authx): add keycloak to the setup launch

* feature (authx): minor formatting for keycloak scripts

DIG-633

* feature (authx): minor formatting for keycloak scripts

DIG-633

* feature (authx): WIP tyk service, simplifying setup

DIG-633

* feature (authx): WIP tyk service, simplifying setup

moving tmp inside lib/tyk alleviates this pain for now but
this is not a good solution as it breaks the repo convention.

DIG-633

* refactor (authx): KEYCLOAK_SERVICE* to KEYCLOAK*

DIG-633

* feature (authx): CHECKPOINT in case of fire

DIG-633

* feature (authx): add + as exclusion in makefile secret generator

DIG-633

* feature (authx): fix tyk confs url

DIG-633

* feature (authx): remove candig-server from authx makefile because it is already launched; add image removal in cleanup; DIG-633

* feature (authx): refactor variables in keycloak script to remove global and rename locals; DIG-633

* feature (authx): formatting; DIG-633

* feature (authx): fix tyk redirect uri instead of candig server in keycloak client redirect uri settings; DIG-633

* feature (authx): add security TODO warning; DIG-633

* feature (authx): CHECKPOINT in case of fire, working on tyk;

DIG-633

* feature (authx): Tyk api redirect works

DIG-633

* docs (authx): document steps, and a todo

DIG-633

* refactor (authx): renames TEMP_KEYCLOAK.. to KEYCLOACL...PROD because thats the purpose of that URL

adds the variable to environment

DIG-633

* feature (authx): analytics for tyk

DIG-633

* feature (authx): remove check for local idp for now

DIG-633

* feature (authx): add warning comments

DIG-633

* feature (authx): add directory cleanup for tyk tmp

DIG-633

* feature (authx): add directory cleanup for tyk tmp

DIG-633

* docs (authx): adds new api section because we need to convey that

right now the tyk setup is adhoc at best, it deploys fine with single
api (candig) but it is not enough. this section documents how to achieve
this in a hacky way. it is rather sad but it is also need of the hour.

DIG-633

* docs (authx): steps to add new api

* feature (authx): resolve conflicts because I clearly cannot read; DIG-633

* feature (authx): add `tee` to logfile

Suggestion at https://github.com/CanDIG/CanDIGv2/pull/99#discussion_r736020301

DIG-633

* feature (authx): fix failing incorrect health checks for containers

DIG-633

* chore (authx): bumps up version of tyk and redis

DIG-633

* fix (authx): remove repeat line

https://github.com/CanDIG/CanDIGv2/pull/99#discussion_r746730247

DIG-633

* fix (authx): indentation should be tabs, not 4 spaces in Make

https://github.com/CanDIG/CanDIGv2/pull/99#discussion_r746732718

DIG-633

* fix (authx): missing new line

https://github.com/CanDIG/CanDIGv2/pull/99#discussion_r746745183

DIG-633

* fix (authx): remove arbiter

https://github.com/CanDIG/CanDIGv2/pull/99#discussion_r746992537

DIG-633

amanjeev commented:
I feel that I have not run the tests some of which are related to auth.  Not sure what's the best way to deal with that is.  Otherwise I am good with this.

shaikh-rashid committed:
reorder of git clone and submodule init
